packages/scripts/src/upgrade-tests/voting2/migrationUtils.ts

```diff
@@ -165,3 +178,47 @@ export const isContractInstance = async (address: string, functionSignature: str
   const encodedSignature = hre.web3.eth.abi.encodeFunctionSignature(functionSignature).slice(2);
   return code.includes(encodedSignature);
 };
+
+export const isGovernorV2Instance = async (address: string): Promise<boolean> => {
+  return isContractInstance(address, "propose((address,uint256,bytes)[],bytes)");
+};
+
+export const isProposerV1Instance = async (address: string): Promise<boolean> => {
+  return isContractInstance(address, "propose((address,uint256,bytes)[])");
+};
+
+export const isVotingV2Instance = async (address: string): Promise<boolean> => {
+  return await isContractInstance(address, "stake(uint256)");
+};
+
+export const proposeEmergency = async (
+  emergencyProposerAddress: string,
+  votingToken: VotingTokenEthers,
+  adminProposalTransactions: AdminProposalTransaction[],
+  afterProposeCallback?: () => void
```

Comment:
nit: just OOC, why use a callback here? Couldn't the caller just use .then() on the returned promise?

Reply:
You are right, `then()` is better. Thank you!